src/main/kotlin/app/batch/S3StreamingWriter.kt

```diff
@@ -63,69 +59,64 @@ class S3StreamingWriter(private val cipherService: CipherService,
             val filePrefix = filePrefix()
             val slashRemovedPrefix = exportPrefix.replace(Regex("""/+$"""), "")
-            val objectKey: String = "${slashRemovedPrefix}/$filePrefix-%06d.txt.${compressionInstanceProvider.compressionExtension()}.enc".format(currentBatch)
-
-            if (!closed) {
-                logger.error("Failed to close output streams cleanly", "object_key" to objectKey)
-            }
-
-            val metadata = ObjectMetadata().apply {
-                contentType = "binary/octetstream"
-                addUserMetadata("x-amz-meta-title", objectKey)
-                addUserMetadata("iv", currentOutputStream!!.initialisationVector)
-                addUserMetadata("cipherText", currentOutputStream!!.dataKeyResult.ciphertextDataKey)
-                addUserMetadata("dataKeyEncryptionKeyId", currentOutputStream!!.dataKeyResult.dataKeyEncryptionKeyId)
-                contentLength = data.size.toLong()
-            }
-
-            logger.info("Putting batch object into bucket",
-                "s3_location" to objectKey, "records_in_batch" to "$recordsInBatch", "batch_size_bytes" to "$batchSizeBytes",
-                "data_size_bytes" to "${data.size}", "export_bucket" to exportBucket, "max_batch_output_size_bytes" to "$maxBatchOutputSizeBytes",
-                "total_snapshot_files_already_written" to "$totalBatches", "total_bytes_already_written" to "$totalBytes",
-                "total_records_already_written" to "$totalRecords")
-
-            inputStream.use {
-                val request = PutObjectRequest(exportBucket, objectKey, it, metadata)
-                s3.putObject(request)
-            }
-
-            logger.info("Put batch object into bucket")
-
-            exportStatusService.incrementExportedCount(objectKey)
-            snapshotSenderMessagingService.notifySnapshotSender(objectKey)
-            
-            totalBatches++
-            totalBytes += batchSizeBytes
-            totalRecords += recordsInBatch
-
-            try {
-                streamingManifestWriter.sendManifest(s3, currentOutputStream!!.manifestFile, manifestBucket, manifestPrefix)
-                totalManifestFiles++
-                totalManifestRecords += currentOutputStream!!.manifestFile.length()
-            } catch (e: Exception) {
-                logger.error("Failed to write manifest", e, "manifest_file" to "${currentOutputStream!!.manifestFile}")
+            val objectKey =
+                "${slashRemovedPrefix}/$filePrefix-%06d.txt.${compressionInstanceProvider.compressionExtension()}.enc"
+                    .format(currentBatch)
+
+            currentOutputStream?.let { encryptingOutputStream ->
+                try {
+                    val closed = encryptingOutputStream.close()
+                    val data = encryptingOutputStream.data()
+
+                    if (!closed) {
+                        logger.error("Failed to close output streams cleanly", "object_key" to objectKey)
```

Comment:
should it also take an action if it fails to close? I didn't see a test for a "failed to close" scenario

Reply:
That's not actually new code - you can see it on line 90 in the old version. I just introduced the let block to get rid of the '!!' expressions, but this introduced a bunch of whitespace only changes.